### PR TITLE
feat: add MenuBar.close()

### LINKED ISSUE #5742: Add `MenuBar.close()`
### Describe your motivation

A PopupButton (a Button which shows a popup window with some rich layout on click) can easily be implemented via a MenuBar:

```java
@Route("")
public class MainView extends VerticalLayout {

    public MainView() {
        final MenuBar menuBar = new MenuBar();
        final SubMenu fooSubMenu = menuBar.addItem("Foo").getSubMenu();
        final HorizontalLayout horizontalLayout = new HorizontalLayout(new Span("Popup button contents"));
        horizontalLayout.add(new Button("OK", e ->
                System.out.println("Clicked")));
        fooSubMenu.add(horizontalLayout);
        add(menuBar);
    }
}
```

The only problem is that, upon clicking on the "OK" button, the submenu stays open.

### Describe the solution you'd like

 It would be great to have a `MenuBar.close()` function which programmatically closes all open submenus. An implementation can be taken from https://github.com/vaadin/web-components/issues/728 : we can simply call `menuBar.getElement().executeJs("this._subMenu.close()");` from the server-side:

```java
@Route("")
public class MainView extends VerticalLayout {

    public MainView() {
        final MenuBar menuBar = new MenuBar();
        final SubMenu fooSubMenu = menuBar.addItem("Foo").getSubMenu();
        final HorizontalLayout horizontalLayout = new HorizontalLayout(new Span("Popup button contents"));
        horizontalLayout.add(new Button("OK", e -> {
            System.out.println("Clicked");
            close(menuBar);
        }));
        fooSubMenu.add(horizontalLayout);
        add(menuBar);
    }

    private static void close(MenuBar menuBar) {
        menuBar.getElement().executeJs("this._subMenu.close()");
    }
}
```

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by mvysny**
## Description

add `MenuBar.close()` which allows us to build a PopupButton easily on top of MenuBar. Also allows you to have menu items different than `<vaadin-menu-item>`, which also close the MenuBar upon clicking (requires a manual click listener + calling `close()`).

Fixes #5742 

## Type of change

- [ ] Bugfix
- [x] Feature

## Checklist

- [x] I have read the contribution guide: https://vaadin.com/docs/latest/contributing/overview
- [x] I have added a description following the guideline.
- [x] The issue is created in the corresponding repository and I have referenced it.
- [x] I have added tests to ensure my change is effective and works as intended.
- [x] New and existing tests are passing locally with my change.
- [x] I have performed self-review and corrected misspellings.
- [ ] I have not completed some of the steps above and my pull request can be closed immediately.

#### Additional for `Feature` type of change

- [x] Enhancement / new feature was discussed in a corresponding GitHub issue and Acceptance Criteria were created.

### CODE REVIEW

File: vaadin-menu-bar-flow-parent/vaadin-menu-bar-flow/src/test/java/com/vaadin/flow/component/menubar/tests/MenuBarTest.java [hunk 1]
```diff
@@ -100,4 +100,9 @@ private void assertChildrenAndItems(MenuItem... expected) {
         menuItems = menuBar.getItems().toArray();
         Assert.assertArrayEquals(expected, menuItems);
     }
+
+    @Test
+    public void close_smoke() {
+        menuBar.close();
```
Comment (tomivirkki) [inline vaadin-menu-bar-flow-parent/vaadin-menu-bar-flow/src/test/java/com/vaadin/flow/component/menubar/tests/MenuBarTest.java:106]: Needs an actual integration test. This test would pass even if the `close`-method was empty.

File: vaadin-menu-bar-flow-parent/vaadin-menu-bar-flow/src/main/java/com/vaadin/flow/component/menubar/MenuBar.java [hunk 2]
```diff
@@ -514,4 +514,11 @@ public void setTooltipText(MenuItem item, String tooltipText) {
 
         item.getElement().setProperty("tooltip", tooltipText);
     }
+
+    /**
+     * Closes any open submenus.
+     */
+    public void close() {
+        getElement().executeJs("this._subMenu.close()");
```
Comment (tomivirkki) [inline vaadin-menu-bar-flow-parent/vaadin-menu-bar-flow/src/main/java/com/vaadin/flow/component/menubar/MenuBar.java:522]: nit: accessing protected API here. Ideally, the API would first be added to the web component and then we'd use the public API here with `this.close()` (btw `this._close()` is already also available).

Comment (mvysny) [reply vaadin-menu-bar-flow-parent/vaadin-menu-bar-flow/src/main/java/com/vaadin/flow/component/menubar/MenuBar.java:522]: I agree. There's https://github.com/vaadin/web-components/issues/728 but it has been dormant for 3 years. I propose to go with the workaround for now.

File: vaadin-menu-bar-flow-parent/vaadin-menu-bar-flow/src/test/java/com/vaadin/flow/component/menubar/tests/MenuBarTest.java (hunk 1 above)
Comment (mvysny) [reply vaadin-menu-bar-flow-parent/vaadin-menu-bar-flow/src/test/java/com/vaadin/flow/component/menubar/tests/MenuBarTest.java:106]: I agree. Let me try to cobble up a TestBench test

Comment (mvysny) [reply vaadin-menu-bar-flow-parent/vaadin-menu-bar-flow/src/test/java/com/vaadin/flow/component/menubar/tests/MenuBarTest.java:106]: Added the `MenuBarPopupButtonIT` test which tests the close() function.

**Review by tomivirkki — APPROVED** [review]

File: vaadin-menu-bar-flow-parent/vaadin-menu-bar-flow-integration-tests/src/main/java/com/vaadin/flow/component/menubar/tests/MenuBarPopupButtonPage.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+ * the License.
+ */
+package com.vaadin.flow.component.menubar.tests;
+
+import com.vaadin.flow.component.contextmenu.SubMenu;
+import com.vaadin.flow.component.html.Div;
+import com.vaadin.flow.component.html.NativeButton;
+import com.vaadin.flow.component.html.Span;
+import com.vaadin.flow.component.menubar.MenuBar;
+import com.vaadin.flow.component.shared.TooltipConfiguration;
```
Comment (tomivirkki) [inline vaadin-menu-bar-flow-parent/vaadin-menu-bar-flow-integration-tests/src/main/java/com/vaadin/flow/component/menubar/tests/MenuBarPopupButtonPage.java:23]: nit: unused import

File: vaadin-menu-bar-flow-parent/vaadin-menu-bar-flow-integration-tests/src/test/java/com/vaadin/flow/component/menubar/tests/MenuBarPopupButtonIT.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    public void init() {
+        open();
+    }
+
+    @Test
+    public void clickingCloseButton_closesSubmenu() {
+        var menuBar = $(MenuBarElement.class).first();
+        menuBar.getButtons().get(0).click();
+        verifyOpened();
+        click("close-button");
```
Comment (tomivirkki) [inline vaadin-menu-bar-flow-parent/vaadin-menu-bar-flow-integration-tests/src/test/java/com/vaadin/flow/component/menubar/tests/MenuBarPopupButtonIT.java:38]: nit: there's a built-in helper `clickElementWithJs` for this

**Review by tomivirkki — DISMISSED** [review]
Blocking merge for now since this might still need some more internal discussion

File: vaadin-menu-bar-flow-parent/vaadin-menu-bar-flow-integration-tests/src/main/java/com/vaadin/flow/component/menubar/tests/MenuBarPopupButtonPage.java (hunk 3 above)
Comment (mvysny) [reply vaadin-menu-bar-flow-parent/vaadin-menu-bar-flow-integration-tests/src/main/java/com/vaadin/flow/component/menubar/tests/MenuBarPopupButtonPage.java]: Done, thanks!

File: vaadin-menu-bar-flow-parent/vaadin-menu-bar-flow-integration-tests/src/test/java/com/vaadin/flow/component/menubar/tests/MenuBarPopupButtonIT.java (hunk 4 above)
Comment (mvysny) [reply vaadin-menu-bar-flow-parent/vaadin-menu-bar-flow-integration-tests/src/test/java/com/vaadin/flow/component/menubar/tests/MenuBarPopupButtonIT.java]: Thanks for the tip, refactored.

**Comment by sissbruecker** [comment]
PR for adding public API to web component: https://github.com/vaadin/web-components/pull/7102

**Review by sissbruecker — APPROVED** [review]

**Pull request merged**